pkg/netpol/eval/exposure.go

```diff
@@ -26,71 +28,53 @@ import (
 
 // this file contains eval.PolicyEngine funcs which are related to exposure-analysis feature
 
-// addRepresentativePods adds representative pods by inferring from network policies selectors within rules
-// the required entities to represent (namespaces or pods by certain labels selectors)
-// for example, if a rule within policy has namespace selector "name: foo", then a representative pod in such a
-// namespace with those labels will be added, representing all potential pods in such a namespace
-func (pe *PolicyEngine) addRepresentativePods() error {
-	// scan policies's rules : generate fake-exposure pod for rules with no match in the resources
-	for _, nsNetpolsMap := range pe.netpolsMap {
-		for pName, policy := range nsNetpolsMap {
-			// scan and handle policy rules which doesn't have a match in the resources
-			err := pe.addPodsForUnmatchedRules(pName, policy)
-			if err != nil {
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-func (pe *PolicyEngine) addPodsForUnmatchedRules(policyName string, policy *k8s.NetworkPolicy) error {
-	// gets the namespaceSelector-s from policy's rules which contain only namespaceSelector
-	namespacesOnly := policy.GetRulesSelectors()
-	// find if there are matches for the namespacesSelectors in pe.namespacesMap; if not add relevant pods
-	err := pe.addPodsForUnmatchedNamespaceSelectors(namespacesOnly, policyName)
-	// TODO : add pods for unmatched rules with:  only podSelector (in the policy's ns)
-	// - namespaceSelector + podSelector
-	return err
-}
-
 // any fake namespace added will start with following prefix for ns name and following pod name
 const repNsNamePrefix = "representative-namespace-"
 
-// gets a list of policy xgress rules consisted only from namespaceSelector.
-// adds new pod for each selector that does not have a matching namespace in the resources
-func (pe *PolicyEngine) addPodsForUnmatchedNamespaceSelectors(nsSelectors []*metav1.LabelSelector, policyName string) error {
-	for i, selector := range nsSelectors {
-		selectorMap, err := metav1.LabelSelectorAsMap(selector)
+// generateRepresentativePeers : generates and adds to policy engine representative peers where each peer
+// has namespace and pod labels inferred from single policy rule labels in the given list of selectors;
+// for example, if a rule within policy has namespaceSelector "foo: managed", then a representative pod in such a
+// namespace with those labels will be added, representing all potential pods in such a namespace.
+// generated representative peers are unique; i.e. if different rules (e.g in different policies or different directions) has same labels :
+// one representative peer is generated to represent both
+func (pe *PolicyEngine) generateRepresentativePeers(selectorsLabels []k8s.SingleRuleLabels, policyName string) error {
+	for i := range selectorsLabels {
+		_, err := pe.AddPodByNameAndNamespace(k8s.RepresentativePodName, repNsNamePrefix+policyName+fmt.Sprint(i), &selectorsLabels[i])
```

Comment:
are all representative pods going to have the same name?
could have a case of 2 such pods in the same namespace but with different pod selectors? 

also, move to another function the code that returns a namespace string for new representative peer.

Reply:
in this case (supporting only ns-selector ) only one pod will be created in each new ns; as each different rule have different namespace labels
so focused only on differentiating the namespace name 

kept "differentiating" the podName for the PR which will support podSelector (supporting both podSelector and nsSelector may require some changes to the for loop)

Reply:
ok, please add a todo comment that this will be changed when supporting pod label selector as well

Reply:
done